pygfx/geometries/_base.py

```diff
@@ -116,12 +123,12 @@ def get_bounding_box(self):
             no finite positions.
 
         """
-        if hasattr(self, "positions"):
-            if self._aabb_rev == self.positions.rev:
+        if isinstance(positions := getattr(self, "positions", None), Buffer):
```

Comment:
want to draw your attention here, since this will affect runtime behavior (but is more strictly accurate).

One thing I've generally struggled with when using pygfx is having a good sense for what the attributes are.  having special attribute names that *may* be present on a Geometry, and must have a specific type when present, definitely makes for difficulty not only in typing, but in interpretability. 

Just flagging that for now, curious whether it's a pattern you're fully settled and happy with for reasons of flexibility/performance, or are open for refactors

Reply:
Doesn't isinstance raise an error when you pass None though?

Reply:
No, it’s fine to ask whether None is an instance of Buffer (it’s not an error).  issubclass however will raise an error of the first argument is not a type.

That said, I have no problem going back.  The main difference here is that you would get a type error on main if positions isn’t a buffer, while here it would just do nothing (silently!). It’s fine to leave the type error and remove this guard if you prefer

Reply:
> having special attribute names that _may_ be present on a Geometry, and must have a specific type when present, definitely makes for difficulty not only in typing, but in interpretability

Would you mind sharing if you think there's any (specific) element(s) of the docs that we should improve in this regard?

Reply:
I understand the "fuzzyness" of the geometry object. It may be worth discussing the idea of producing a standardized list of attributes that a geometry commonly has, and their types and shape. We could even make real properties for them? Though it should always be possible for custom objects to use new geometry attributes. Let's discuss in a separate issue.

> Many attributes of Trackables have no "go-to-definition" ability

I was not aware that. But this we can fix with type annotations, correct?

Reply:
FWIW somethings I like to think of the geometry as just a dict with a bunch of buffers and textures 🤷

Reply:
sounds good.

python can definitely express "if a key exists, it has this type" with TypedDict.  Having the attribute access as a shortcut makes things a bit harder.  But there are a few things we can do.  I have another branch working on Resource and will open a new PR to discuss those things